code_generation/quantity.py

```diff
@@ -1,11 +1,14 @@
 import logging
+import time
 
 log = logging.getLogger(__name__)
 
 
 class Quantity:
     def __init__(self, name):
         self.name = name
+        # structure for storing shifts:
+        # {scope1: {shift1 : name2}, {shift2: name2}, scope2: {shift1 : name2}, {shift2: name2}, ...}
```

Comment:
better use python dict syntax which was maybe intended anyway (remove brackes between shift1 and shift2)